src/common/image.c

```diff
@@ -935,9 +928,18 @@ int32_t dt_image_duplicate_with_version(const int32_t imgid, const int32_t newve
 
     g_free(filename);
 
-    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, imgid, 'r');
+    const dt_image_t *img = dt_image_cache_get(darktable.image_cache, newid, 'r');
```

Comment:
This looks wrong to me. You are setting the group-id to be the one of the new image (the duplicate itself). The group-id of the artifact should be the group of the main image.

Reply:
That makes sense. I'll fix/re-test...

Reply:
Note that the issue may already be fixed with my latest work on duplicate.

Reply:
I'm just looking at this. This PR fixes something different, that the calls to dt_tag_detach_by_string() and dt_image_cache_unset_change_timestamp() will call dt_image_cache_unset_change_timestamp() which load the version # in from the DB. So it's important to set the version # before it gets cached. (Or an alternative would be to make the version # writeable via the cache.)

Also dt_image_cache_unset_change_timestamp() should be working on the new image, even if I was wrong about the group ID.